From bfc7a3a868779bd23625ba385997ab92de9f4047 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Thu, 9 Jul 2020 22:58:39 +0300
Subject: [PATCH 54/56] mcm-imx8m-mini: Add board_usb_init() and
 board_usb_cleanup()

Enable power on/off usb power in U-Boot.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 board/compulab/mcm-imx8m-mini/mcm-imx8m-mini.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/board/compulab/mcm-imx8m-mini/mcm-imx8m-mini.c b/board/compulab/mcm-imx8m-mini/mcm-imx8m-mini.c
index 9ece28b318..6cea3b38cd 100644
--- a/board/compulab/mcm-imx8m-mini/mcm-imx8m-mini.c
+++ b/board/compulab/mcm-imx8m-mini/mcm-imx8m-mini.c
@@ -267,6 +267,23 @@ int board_phy_config(struct phy_device *phydev)
 }
 #endif
 
+int board_usb_init(int index, enum usb_init_type init)
+{
+	debug("board_usb_init %d, type %d\n", index, init);
+
+	imx8m_usb_power(index, true);
+
+	return 0;
+}
+
+int board_usb_cleanup(int index, enum usb_init_type init)
+{
+	debug("board_usb_cleanup %d, type %d\n", index, init);
+
+	imx8m_usb_power(index, false);
+
+	return 0;
+}
 int board_init(void)
 {
 
-- 
2.11.0

